### ISSUE TITLE
Article idea: Javascript styleguides (not only CSS...)

### ISSUE THREAD

**Issue by franciov** [operasoftware/devopera#362]
The article would be about why it's important to follow a JS styleguide

It will be based on this AngularJS styleguide: https://github.com/johnpapa/angular-styleguide

What are your thoughts?

**Comment by shwetank** [comment]
I think a comparision of various styleguides and what are interesting about each (in isolation and in comparision to each other) would be great. There is [AirBNB](https://github.com/airbnb/javascript)'s stylegude, Feross's [Standard Stylegude](https://github.com/feross/standard) and many more.

**Comment by franciov** [comment]
@shwetank I can write about that, focusing on pure Javascript, and only mentioning frameworks maybe.

Here is a list of Javascript styleguides:

- [Google Javascript Guide](https://google.github.io/styleguide/javascriptguide.xml)
- [Mozilla coding style](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style)
- [Crockford’s code conventions](http://javascript.crockford.com/code.html)
- [Waldron’s Principles of Writing Consistent, Idiomatic JavaScript](https://github.com/rwaldron/idiomatic.js/)
- [Github coding style for Javascript](https://github.com/styleguide/javascript)
- [npm’s “funny” coding style](https://docs.npmjs.com/misc/coding-style)
- [jQuery’s Javascript Style Guide](https://contribute.jquery.org/style-guide/js/)
- [Feross' Javascript Standard Styleguide](https://github.com/feross/standard)
- [Airbnb javascript style guide](https://github.com/airbnb/javascript)

I can write about each one, and compare the best ones in terms of style and tools provided.

**Comment by shwetank** [comment]
Yes, that sounds great. If we can have an article which stays mostly neutral but at the same time provides a comparision of the different philosophies of each styleguide, then that would make for a great article.

**Comment by franciov** [comment]
@shwetank I'm gonna start working on it soon then, further hints are appreciated meanwhile.
